proposals/0929-legacy-architecture-removal.md

```diff
@@ -0,0 +1,347 @@
+    * We stopped accepting PRs that modify the Legacy Architecture.
+    * We stopped testing the Legacy Architecture when releasing React Native.
+* **2025-06-12: [React Native 0.80]** Legacy Architecture [warnings are added](https://reactnative.dev/blog/2025/06/12/react-native-0.80#legacy-architecture-freezing--warnings)
+    * Users will see a warning on DevTools when using Legacy Architecture.
+    * Moreover, specific warnings have been added to notify the user if they’re using a pattern that won’t be supported by interop layers
+* **2025-08-12: [React Native 0.81]** Further Legacy Architecture classes are marked as `@Deprecated`
+    * Users will be notified at build time if using a `@Deprecated` class through a build warning.
+
+### Upcoming steps
+
+* **2025-10-x: [React Native 0.82]** New Architecture opt-out is removed.
+    * Users won’t be able to opt out of the newArchitecture anymore.
+    * Specifically:
+        * On Android, setting `newArchEnabled=false` will result in a warning message and the property will be hardcoded to `true`
+        * To keep compatibility with libraries, we will set `newArchEnabled=true` if a user removes the `newArchEnabled` line from their `gradle.properties`.
+        * On iOS
+            * Installing pods with `RCT_NEW_ARCH_ENABLED=0` will result in a warning message and the property will be hardcoded to `1`
+            * Changing the `RCTNewArchEnabled` flag in the `Info.plist` will be ignored
+            * Overriding methods such as `newArchEnabled` in the `AppDelegate` will be ignored
+        * To keep compatibility with libraries, we will set `RCT_NEW_ARCH_ENABLED=1`in all the podspecs, including the one autolinked. This will give library authors the time to remove the legacy arch code from their library.
+    * We don’t forecast removal of Legacy Architecture APIs in this version (i.e. Legacy Architecture will still be inside React Native core, but not accessible).
+* **2025-12-x: [React Native 0.83]** Legacy Architecture internal and non-public classes are removed
+    * By this stage, we will start removing parts of the Legacy Architecture from the core of React Native (see list below).
+    * In order to simplify the adoption, we won’t be removing the Interop Layer at all (see paragraph below - ‘The future of the Interop Layer’)
+    * Also, in order to simplify the migration, we won’t be removing any public classes from Legacy Architecture at this point. This is to support libraries and modules that might still reference Legacy 
+* **2025-12-x: Expo SDK 55**
+    * We expect Expo SDK 55 to ship React Native 0.83. This is going to be the first version of Expo where users can’t opt out from the New Architecture.
+* **2026-02-x: [React Native 0.84]** Legacy Architecture public classes will start to be removed
+    * From 0.84, we will start removing classes and APIs from the Legacy Architecture. See below for the list of APIs we intend to remove.
```

Comment:
❓ Question:

For someone less familiar with the internals of RN, those two lines can be ambiguous wrt the Interop layer.
In the sense that we could wonder if the Interop layer is using those internal and public classes.

Does "Legacy Architecture public classes will start to be removed" impact the interop layer?

I think you clarify this below by saying 
> Therefore, we’ll keep the Legacy Architecture APIs used by the Interop Layer to make them work properly.

but I am not sure

Reply:
> Does "Legacy Architecture public classes will start to be removed" impact the interop layer?

Nope it does not involve classes that are used or needed by the interop layers. I can add a sentence to clarify this.

Reply:
thanks 🙏 all clear